AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1793,7 +1793,8 @@ - (BOOL)placeholderShouldPersist
 
 - (BOOL)_shouldHavePlaceholderLayer
 {
-  return (_placeholderEnabled && [self __implementsDisplay]);
+  CGSize layoutSize = (_layout ? _layout.size : CGSizeZero);
```

Comment:
What other size would we check? `calculatedSize` is the same value, which is what a manually sized node would use for the final frame size.